[workaround available] Java 24 compatibility

# Issue: Incompatibility with Java 24

Tzatziki is not compatible with Java 24.
Every test execution fails with the following error:
```
java.lang.ExceptionInInitializerError
	at com.decathlon.tzatziki.utils.Fields.recursivelyCacheFieldsOf(Fields.java:83)
	at com.decathlon.tzatziki.utils.Fields.recursivelyCacheFieldsOf(Fields.java:72)
	at com.decathlon.tzatziki.utils.Fields.lambda$byName$0(Fields.java:59)
	at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1229)
	at java.base/java.util.Collections$SynchronizedMap.computeIfAbsent(Collections.java:2981)
	at com.decathlon.tzatziki.utils.Fields.byName(Fields.java:59)
	at com.decathlon.tzatziki.utils.Fields.getField(Fields.java:55)
	at com.decathlon.tzatziki.utils.Env.export(Env.java:26)
	at com.decathlon.tzatziki.utils.EnvTest.export(EnvTest.java:14)
	at java.base/java.lang.reflect.Method.invoke(Method.java:565)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
Caused by: java.lang.RuntimeException: io.github.toolfactory.jvm.Driver$InitializeException: Could not initialize DynamicDriver
	at org.burningwave.core.assembler.StaticComponentContainer.<clinit>(StaticComponentContainer.java:481)
	... 12 more
Caused by: io.github.toolfactory.jvm.Driver$InitializeException: Could not initialize DynamicDriver
	at io.github.toolfactory.jvm.DriverAbst.refresh(DriverAbst.java:257)
	at io.github.toolfactory.jvm.DriverAbst.getBuiltinClassLoaderClass(DriverAbst.java:944)
	at org.burningwave.core.classes.Classes$Loaders.<init>(Classes.java:346)
	at org.burningwave.core.classes.Classes$Loaders.create(Classes.java:359)
	at org.burningwave.core.assembler.StaticComponentContainer.<clinit>(StaticComponentContainer.java:382)
	... 12 more
Caused by: io.github.toolfactory.jvm.util.ObjectProvider$BuildingException: Exception occurred while retrieving the implementation of class io.github.toolfactory.jvm.function.catalog.GetResourcesFunction (jvm architecture: x64, jvm version: 24, jvm vendor: Eclipse Adoptium)
	at io.github.toolfactory.jvm.util.ObjectProvider.getOrBuildObject(ObjectProvider.java:95)
	at io.github.toolfactory.jvm.DriverAbst.getOrBuildResourcesRetriever(DriverAbst.java:436)
	at io.github.toolfactory.jvm.DriverAbst.refresh(DriverAbst.java:229)
	... 16 more
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:74)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:483)
	at io.github.toolfactory.jvm.util.ObjectProvider.getOrBuildObjectInternal(ObjectProvider.java:144)
	at io.github.toolfactory.jvm.util.ObjectProvider.getOrBuildObject(ObjectProvider.java:92)
	... 18 more
Caused by: java.lang.NoSuchMethodException: no such method: jdk.internal.loader.URLClassPath.findResources(String,boolean)Enumeration/invokeVirtual
	at java.base/java.lang.invoke.MemberName.makeAccessException(MemberName.java:910)
	at java.base/java.lang.invoke.MemberName$Factory.resolveOrFail(MemberName.java:989)
	at java.base/java.lang.invoke.MethodHandles$Lookup.resolveOrFail(MethodHandles.java:3592)
	at java.base/java.lang.invoke.MethodHandles$Lookup.findVirtual(MethodHandles.java:2611)
	at io.github.toolfactory.jvm.function.catalog.GetResourcesFunction$ForJava9$1.<init>(GetResourcesFunction.java:164)
	at io.github.toolfactory.jvm.function.catalog.GetResourcesFunction$ForJava9.buildResourceFinder(GetResourcesFunction.java:134)
	at io.github.toolfactory.jvm.function.catalog.GetResourcesFunction$Abst.<init>(GetResourcesFunction.java:56)
	at io.github.toolfactory.jvm.function.catalog.GetResourcesFunction$ForJava7.<init>(GetResourcesFunction.java:92)
	at io.github.toolfactory.jvm.function.catalog.GetResourcesFunction$ForJava9.<init>(GetResourcesFunction.java:123)
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
	... 22 more
Caused by: java.lang.NoSuchMethodError: method resolution failed
	at java.base/java.lang.invoke.MethodHandleNatives.resolve(Native Method)
	at java.base/java.lang.invoke.MemberName$Factory.resolve(MemberName.java:957)
	at java.base/java.lang.invoke.MemberName$Factory.resolveOrFail(MemberName.java:986)
	... 30 more
```

# Steps to reproduce

You can reproduce this issue in two ways:

1. Run any Tzatziki test in an application using Java 24.
1. Attempt to build Tzatziki using Java 24.

**Impacted Version**: Tzatziki 1.8.2 and earlier.

# Cause

The issue arises because `tzatziki-common` depends on the `io.github.toolsfactory:jvmdriver` library through the `org.burningwave:core library`. Unfortunately, `jvmdriver` is [not compatible with Java 24](https://github.com/toolfactory/jvm-driver/issues/71).

# Proposed solutions

## Plan A: Switch Libraries
Migrate Tzatziki to an alternative library that is compatible with Java 24.

## Plan B: Wait for a Fix
Wait for the `jvmdriver` library to be fixed ([track issue here](https://github.com/toolfactory/jvm-driver/issues/71)) or contribute a pull request to address the incompatibility.

--- a/tzatziki-common/pom.xml
+++ b/tzatziki-common/pom.xml
@@ -136,7 +136,7 @@
         <dependency>
             <groupId>org.burningwave</groupId>
             <artifactId>core</artifactId>
-            <version>12.65.2</version>
+            <version>12.66.1</version>
         </dependency>
         <dependency>
             <groupId>com.decathlon.tzatziki</groupId>

